citesphere/src/main/java/edu/asu/diging/citesphere/web/user/UpdateItemController.java

```diff
@@ -0,0 +1,67 @@
+package edu.asu.diging.citesphere.web.user;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.social.zotero.exception.ZoteroConnectionException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.citesphere.core.exceptions.AccessForbiddenException;
+import edu.asu.diging.citesphere.core.exceptions.CannotFindCitationException;
+import edu.asu.diging.citesphere.core.exceptions.CitationIsOutdatedException;
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroHttpStatusException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroItemCreationFailedException;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.model.bib.ICitation;
+import edu.asu.diging.citesphere.user.IUser;
+
+@Controller
+public class UpdateItemController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    @RequestMapping(value = "/auth/group/{zoteroGroupId}/items/{itemId}/update", method = RequestMethod.POST)
+    public ResponseEntity<?> updateReference(Authentication authentication,
+            @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId,
+            @RequestParam(value = "referenceCitationKey") String referenceCitationKey,
+            @RequestParam(value = "reference") String reference) {
+        try {
+            ICitation citation = citationManager.getCitation((IUser) authentication.getPrincipal(), zoteroGroupId,
+                    itemId);
+            citation = citationManager.updateCitationReference(citation, referenceCitationKey, reference);
+            citationManager.updateCitation((IUser) authentication.getPrincipal(), zoteroGroupId, citation);
```

Comment:
what do `updateCitationReference` and `updateCitation` do? why do they both need to be called? if they do things that should always be done together, then this is business logic and should not be in the controller.

Reply:
updateCitation method is used at multiple places so, didn't change it. The method saves changed citation data to zetero.  updateCitationReference name is changed to addCitationToReferences, which adds found citation to references of the editing citation.